File: EXE201_Heureux/src/main/java/com/example/exe201_heureux/entity/Report.java
package com.example.exe201_heureux.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.Instant;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "Report")
public class Report {
    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teamid")
    private Team teamid;

    @Column(name = "create_date")
    private Instant createDate;

    @Lob
    @Column(name = "description")
    private String description;

    @Size(max = 50)
    @Column(name = "status", length = 50)
    private String status;

}

File: Apis/Domain/Entities/Report.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities;

public class Report : BaseEntity
{

    public int? Teamid { get; set; }

    public DateTime? CreateDate { get; set; }

    public string? Status { get; set; }

    public string? Description { get; set; }

    public virtual Team? Team { get; set; }
}
